controllers/applications.js

```diff
@@ -1,44 +1,39 @@
-const fs = require('fs');
-const Applications = require('../data/applications.json');
+const Applications = require('../models/Applications');
+
+const listApplication = (req, res) => {
+  Applications.find(req.query)
+    .then((application) => res.status(200).json(application))
+    .catch((error) => res.status(404).json({ message: error }));
+};
 
 const createApplication = (req, res) => {
-  const newApplication = {
-    id: new Date().getTime().toString(),
-    idPosition: req.query.idPosition,
-    idPostulant: req.query.idPostulant,
-    date: req.query.date,
-    profile: req.query.profile,
-  };
-  Applications.push(newApplication);
-  fs.writeFile('./data/applications.json', JSON.stringify(Applications), {}, (err) => {
-    if (err) {
-      return res.status(400).send(err);
+  const application = new Applications({
+    positions: req.body.positions,
+    postulants: req.body.postulants,
+    interview: req.body.interview,
+    result: req.body.result,
+  });
+
+  application.save((error, app) => {
+    if (error) {
+      return res.status(400).json(error);
     }
-    return res.status(201).send(newApplication);
+    return res.status(201).json(app);
   });
 };
 
 const deleteApplication = (req, res) => {
-  let removedApplication;
-  const filteredApplications = Applications.filter((app) => {
-    if (app.id === req.query.id) {
-      removedApplication = app;
-      return false;
+  Applications.findByIdAndDelete(req.params.id, (error, chosenApplication) => {
+    if (!chosenApplication) {
+      return res.status(404).json({ message: `Id ${req.params.id} does not exist` });
     }
-    return true;
-  });
-
-  if (!removedApplication.length) res.status(404).send('application not found');
-  fs.writeFile('./data/applications.json', JSON.stringify(filteredApplications), {}, (err) => {
-    if (err) {
-      return res.status(400).send(err);
+    if (error) {
+      return res.status(400).json(error);
     }
-    return res.status(204).send(removedApplication);
+    return res.status(204).send({ message: `Id ${req.params.id} was remove successfully` });
```

Comment:
status 204 don't send messages

Reply:
In this case, should I send a 204 for the delete or a 200 with the message??